src/app/services/todo/todo.service.impl.ts

```diff
@@ -0,0 +1,18 @@
+import { TodoService } from "./todo.service";
+
+export default class TodoServiceImpl implements TodoService {
+  addTodo(name: string): Promise<void> {
+    let data = localStorage.getItem("todo");
+    /****  Stringify setItem values that takes string or null  ****/
+    if (data) {
+      let dataArr = JSON.parse(data);
+      dataArr.push(name);
```

Comment:
We should store id along with name. You would need that to delete etc.

Reply:
Consider creating a class

```
class TodoItem {
  id
  name
  
  serialize() { // returns json {id: name:}
  }
}
```

Reply:
we can access id through dataArr.